ArduPlane/tailsitter.cpp

```diff
@@ -85,44 +85,49 @@ void QuadPlane::tailsitter_output(void)
             throttle = motors->get_throttle_hover() * 100;
             SRV_Channels::set_output_scaled(SRV_Channel::k_rudder, 0);
             pos_control->get_accel_z_pid().set_integrator(throttle*10);
-            
-            if (mask == 0) {
-                // override AP_MotorsTailsitter throttles during back transition
-                SRV_Channels::set_output_scaled(SRV_Channel::k_throttle, throttle);
-                SRV_Channels::set_output_scaled(SRV_Channel::k_throttleLeft, throttle);
-                SRV_Channels::set_output_scaled(SRV_Channel::k_throttleRight, throttle);            
-            }
+
+            // override AP_MotorsTailsitter throttles during back transition
+            SRV_Channels::set_output_scaled(SRV_Channel::k_throttle, throttle);
+            SRV_Channels::set_output_scaled(SRV_Channel::k_throttleLeft, throttle);
+            SRV_Channels::set_output_scaled(SRV_Channel::k_throttleRight, throttle);
         }
-        if (mask != 0) {
+        if (tailsitter.motor_mask != 0) {
             // set AP_MotorsMatrix throttles enabled for forward flight
-            motors->output_motor_mask(throttle * 0.01f, mask, plane.rudder_dt);
-            }
+            motors->output_motor_mask(throttle * 0.01f, tailsitter.motor_mask, plane.rudder_dt);
         }
         return;
     }
-    
-    // handle VTOL modes
-    // the MultiCopter rate controller has already been run in an earlier call 
-    // to motors_output() from quadplane.update()
-    motors_output(false);
+
+    if (assisted_flight) {
+        control_stabilize();
+        motors_output(true);
```

Comment:
We're calling motors_output more than once in each pass through set_servos, right?
The option to not run the rate controller is to make the multiple calls less inefficient?

Reply:
this is for Q-assist, it we're in a normal vtol mode this is already done in the quadplane code, altough i'm not sure why I need to call control stabalise again